draft/2020-12/release-notes.md

```diff
@@ -2,7 +2,479 @@
+
+Although the meaning of `items` has changed, the syntax for defining arrays
+remains the same. Only the syntax for defining tuples has changed. The idea is
+that an array has items (`items`) and optionally has some positionally defined
+items that come before the normal items (`prefixItems`).
+
+Here are some examples to illustrate the changes.
+
+### Open tuple
+<table>
+  <tr>
+    <th>Draft 2019-09</th>
+    <th>Draft 2020-12</th>
+  </tr>
+  <tr>
+    <td>
+      <pre>{
+  "items": [
+    { "$ref": "#/$defs/foo" },
+    { "$ref": "#/$defs/bar" }
+  ]
+}</pre>
+    </td>
+    <td>
+      <pre>{
+  "prefixItems": [
+    { "$ref": "#/$defs/foo" },
+    { "$ref": "#/$defs/bar" }
+  ]
+}</pre>
+    </td>
+  </tr>
+</table>
+
+### Closed tuple
+<table>
+  <tr>
+    <th>Draft 2019-09</th>
+    <th>Draft 2020-12</th>
+  </tr>
+  <tr>
+    <td>
+      <pre>{
+  "items": [
+    { "$ref": "#/$defs/foo" },
+    { "$ref": "#/$defs/bar" }
+  ],
+  "additionalItems": false
+}</pre>
+    </td>
+    <td>
+      <pre>{
+  "prefixItems": [
+    { "$ref": "#/$defs/foo" },
+    { "$ref": "#/$defs/bar" }
+  ],
+  "items": false
+}</pre>
+    </td>
+  </tr>
+</table>
+
+### Tuple with constrained additional items
+<table>
+  <tr>
+    <th>Draft 2019-09</th>
+    <th>Draft 2020-12</th>
+  </tr>
+  <tr>
+    <td>
+      <pre>{
+  "items": [
+    { "$ref": "#/$defs/foo" },
+    { "$ref": "#/$defs/bar" }
+  ],
+  "additionalItems": { "$ref": "#/$defs/baz" }
+}</pre>
+    </td>
+    <td>
+      <pre>{
+  "prefixItems": [
+    { "$ref": "#/$defs/foo" },
+    { "$ref": "#/$defs/bar" }
+  ],
+  "items": { "$ref": "#/$defs/baz" }
+}</pre>
+    </td>
+  </tr>
+</table>
+
+## $dynamicRef and $dynamicAnchor
+The `$recursiveRef` and `$recursiveAnchor` keywords were replaced by the more
+powerful `$dynamicRef` and `$dynamicAnchor` keywords. `$recursiveRef` and
+`$recursiveAnchor` were introduced in the previous draft to solve the problem of
+extending recursive schemas. As the "recursive" keywords got some use and we
+understood them better, we discovered how we could generalize them to solve even
+more types of problems. The name change reflects that these keywords are useful
+for more than just extending recursive schemas.
+
+A `$dynamicAnchor` can be thought of like a normal `$anchor` except that it can
+be referenced across schemas rather than just in the schema where it was
+defined. You can think of the old `$recursiveAnchor` as working the same way
+except that it only allowed you to create one anchor per schema, it had to be at
+the root of the schema, and the anchor name is always empty.
+
+`$dynamicRef` works the same as the old `$recursiveRef` except that fragments
+are no longer empty (`"$dynamicRef": "#my-anchor"` instead of `"$recursiveRef":
+"#"`) and non-fragment-only URIs are allowed. When a `$dynamicRef` contains a
+non-fragment-only URI-Reference, the schema the URI-Reference resolves to is
+used as the starting point for dynamic resolution.
```

Comment:
"the schema the URI-Reference resolves to is used as the starting point for dynamic resolution" -- I don't think this is right:

> If the initially resolved starting point URI includes a fragment that was created by the "$dynamicAnchor" keyword, ... Otherwise, its behavior is identical to "$ref", and no runtime resolution is needed. 

(https://json-schema.org/draft/2020-12/json-schema-core.html#rfc.section.8.2.3.2)

Reply:
I don't see the problem. What about these two statements are in conflict?

Reply:
The release notes seem to be saying that `"$dynamicRef": "someuri#/foo/bar/baz"` is a valid starting point for dynamic resolution, but in fact URIs of that form can only be treated as if `$ref` was used.

Reply:
I think the disconnect might be in the meaning of "dynamic resolution". The way I've used it, it includes the processes of determining whether the fragment is a usable dynamic anchor. If it isn't, the result of dynamic resolution is to do nothing and it's effective behavior is that of `$ref` (just the initial resolution). Maybe you are considering "dynamic resolution" to mean only be the process after determining that we have a valid dynamic anchor?

Reply:
Yes, I read this as saying that if the target of the initial URI has a $dynamicAnchor present, then the dynamic scope resolution process can continue.

Maybe that entire final sentence can be removed. Implementers need to read the spec fully to understand how to implement this keyword.

Reply:
I agree that it would work if we remove the sentence, but I think it's better if we can provide some brief explanation of what a non-fragment-only URI-Reference would do after telling people that it's allowed. I'd rather try to improve that sentence.

Reply:
Is this document the place for that? (e.g. https://github.com/json-schema-org/json-schema-org.github.io/pull/392/files/dc25d4126cd3f54c770480da5aff584476fc68f8#r643340334)

Reply:
We're just talking about a brief note about what has changed and generally what to expect from that change. That's exactly what this document is for. The comment you reference is about expanding the example to showcase the new functionality. I wouldn't go that far in this case either.